From d818828292ec23e26a682c5c24a77051f1270896 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Sat, 25 Jul 2009 17:36:51 +0200
Subject: [PATCH 1/3] Eat own dog food: use cURL to download CAs

This is because our installation of Perl doesn't include LWP but we do
have a version of curl already available.

Avoid trying to check the certificate when fetching the CA bundle.
As we are using curl and we don't have any CA certificates yet. This
doesn't work if we try checking the certificate chain.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
Signed-off-by: Pat Thoyts <patthoyts@users.sourceforge.net>

diff --git a/lib/mk-ca-bundle.pl b/lib/mk-ca-bundle.pl
index 0b13318..079f45d 100644
--- a/lib/mk-ca-bundle.pl
+++ b/lib/mk-ca-bundle.pl
@@ -32,7 +32,6 @@
 #
 use Getopt::Std;
 use MIME::Base64;
-use LWP::UserAgent;
 use strict;
 use vars qw($opt_b $opt_d $opt_f $opt_h $opt_i $opt_l $opt_n $opt_p $opt_q $opt_s $opt_t $opt_u $opt_v $opt_w);
 use List::Util;
@@ -128,8 +127,6 @@ if ($opt_i) {
   print "Operating System Name            : $^O\n";
   print "Getopt::Std.pm Version           : ${Getopt::Std::VERSION}\n";
   print "MIME::Base64.pm Version          : ${MIME::Base64::VERSION}\n";
-  print "LWP::UserAgent.pm Version        : ${LWP::UserAgent::VERSION}\n";
-  print "LWP.pm Version                   : ${LWP::VERSION}\n";
   print "Digest::SHA.pm Version           : ${Digest::SHA::VERSION}\n" if ($Digest::SHA::VERSION);
   print "Digest::SHA::PurePerl.pm Version : ${Digest::SHA::PurePerl::VERSION}\n" if ($Digest::SHA::PurePerl::VERSION);
   print ("=" x 78 . "\n");
@@ -296,21 +293,7 @@ if($curl && !$opt_n) {
 }
 
 unless ($fetched || ($opt_n and -e $txt)) {
-  my $ua  = new LWP::UserAgent(agent => "$0/$version");
-  $ua->env_proxy();
-  $resp = $ua->mirror($url, $txt);
-  if ($resp && $resp->code eq '304') {
-    print STDERR "Not modified\n" unless $opt_q;
-    exit 0 if -e $crt && !$opt_f;
-  } else {
-      $fetched = 1;
-  }
-  if( !$resp || $resp->code !~ /^(?:200|304)$/ ) {
-      print STDERR "Unable to download latest data: "
-        . ($resp? $resp->code . ' - ' . $resp->message : "LWP failed") . "\n"
-        unless $opt_q;
-      exit 1 if -e $crt || ! -r $txt;
-  }
+  `curl --insecure $url > $txt`;
 }
 
 my $filedate = $resp ? $resp->last_modified : (stat($txt))[9];
-- 
2.1.0

